File: app/src/main/java/com/databits/androidscouting/layout/LayoutConfig.java
package com.databits.androidscouting.layout;

import com.databits.androidscouting.model.Cell;
import java.util.List;

/**
 * Simple wrapper class for JSON layout deserialization.
 * Matches the structure of MultiviewTypeAdapter for JSON compatibility.
 */
public class LayoutConfig {
    public List<Cell> mCell;

    public LayoutConfig() {
        // Default constructor for Moshi
    }

    public LayoutConfig(List<Cell> cells) {
        this.mCell = cells;
    }
}

File: app/src/main/java/com/databits/androidscouting/model/LayoutConfig.kt
package com.databits.androidscouting.model

import com.squareup.moshi.JsonClass

/**
 * Kotlin version of LayoutConfig with Moshi codegen.
 * Top-level wrapper for layout JSON structure.
 */
@JsonClass(generateAdapter = true)
data class LayoutConfig(
    val cells: List<Cell>,
    val version: Int = 1
) {
    /**
     * Validate all cells in the layout.
     * Returns list of validation errors (empty if all valid).
     */
    fun validate(): List<ValidationResult.Error> {
        return cells.mapIndexedNotNull { index, cell ->
            val result = cell.validate()
            if (result is ValidationResult.Error) {
                ValidationResult.Error("Cell $index (${cell.title}): ${result.message}")
            } else {
                null
            }
        }
    }
}
